src/ucp/core/ucp_worker.c

```diff
@@ -2362,6 +2363,7 @@ static unsigned ucp_worker_discard_uct_ep_destroy_progress(void *arg)
                   uct_ep, worker);
     }
 
+    ucp_ep_unprogress(ucp_ep, uct_ep, rsc_index);
```

Comment:
IMO at this point, uct_ep cannot be a wireup_ep (since we already checked it before calling ucp_worker_discard_tl_uct_ep), so to simplify the code, ucp_ep_unprogress could avoid checking for wireup_ep at all.

on other hand, when calling ucp_ep_unprogress from ucp_ep_cleanup_lanes, we still need to take care of unprogressing wireup_ep->super.uct_ep. BUT - we can do this from `UCS_CLASS_CLEANUP_FUNC(ucp_wireup_ep_t)`, like we do for wireup_ep->aux_ep.

Reply:
we don't have `rsc_index `saved for `wireup_ep->super.uct_ep` how do we unprogress `wireup_ep->super.uct_ep` in `UCS_CLASS_CLEANUP_FUNC(ucp_wireup_ep_t)`?

Reply:
how bing change is to add rsc_index of uct_ep to proxy_ep ?

Reply:
did you mean adding `rsc_index `to `ucp_proxy_ep_t`? Or add to `ucp_wireup_ep `in addition to `aux_rsc_index`?

Reply:
proxy_ep - because that's where uct_ep is

Reply:
done